packages/components/src/navigation/stories/components-composition.js

```diff
@@ -0,0 +1,132 @@
+/**
+ * External dependencies
+ */
+import styled from '@emotion/styled';
+
+/**
+ * WordPress dependencies
+ */
+import { Button } from '@wordpress/components';
+import { useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import ComponentsCompositionNavigation, {
+	NavigationItem,
+	NavigationLevel,
+} from '../components-composition-navigation';
+
+export default {
+	title: 'Components/Navigation/Components Composition',
+	component: ComponentsCompositionNavigation,
+};
+
+const Container = styled.div`
+	max-width: 246px;
+`;
+
+function ComponentsCompositionExample() {
+	const [ item, setItem ] = useState( 'item-1' );
+	const [ level, setLevel ] = useState( 'root' );
+
+	return (
+		<Container>
+			<ComponentsCompositionNavigation
+				activeItem={ item }
+				activeLevel={ level }
+				setActiveItem={ setItem }
+				setActiveLevel={ setLevel }
+			>
+				<NavigationLevel title="Home">
+					<NavigationItem item="item-1" title="Item 1" />
+					<NavigationItem item="item-2" title="Item 2" />
+					<NavigationItem
+						badge="2"
+						navigateToLevel="category"
+						item="item-3"
+						title="Category"
+					/>
+					<NavigationItem
+						href="https://wordpress.org/"
+						item="item-4"
+						target="_blank"
+						title="External link"
+					/>
+					<NavigationItem item="item-5">
+						<img
+							alt="WordPress Logo"
+							src="https://s.w.org/style/images/about/WordPress-logotype-wmark-white.png"
+							style={ { width: 50, height: 50 } }
+						/>
+					</NavigationItem>
+				</NavigationLevel>
+
+				<NavigationLevel
+					level="category"
+					parentLevel="root"
+					parentLevelTitle="Home"
+					title="Category"
+				>
+					<NavigationItem badge="1" item="child-1" title="Child 1" />
+					<NavigationItem item="child-2" title="Child 2" />
+					<NavigationItem
+						navigateToLevel="nested-category"
+						item="child-3"
+						title="Nested Category"
+					/>
+				</NavigationLevel>
+
+				<NavigationLevel
+					level="nested-category"
+					parentLevel="category"
+					parentLevelTitle="Category"
+					title="Nested Category"
+				>
+					<NavigationItem item="sub-child-1" title="Sub Child 1" />
+					<NavigationItem item="sub-child-2" title="Sub Child 2" />
+				</NavigationLevel>
+
+				<NavigationLevel title="Secondary Menu">
+					<NavigationItem
+						navigateToLevel="secondary-level"
+						item="secondary-item-1"
+						title="Secondary Item 1"
+					/>
+					<NavigationItem
+						item="secondary-item-2"
+						title="Secondary Item 2"
+					/>
+				</NavigationLevel>
```

Comment:
If this `NavigationLevel` were somewhere other than `root`, (if I follow correctly) it would designate a `parentLevel` prop. This would unfortunately render a back button.

Reply:
I'm probably misunderstanding your comment, but just in case... 😅 

Right now all `NavigationLevel` are supposed to be written "flatly" in the `Navigation` root.

By default they are `level="root"`, meaning that they will be rendered in the initial level of the navigation.
If `level` is not root, they will be hidden from the initial level, and are toggled with `setLevel`.

The optional `parentLevel` is what renders the back button.
Hypothetically, we could create a non-root level _without_ back button like this:
```jsx
<NavigationLevel level="foobar" title="Nested Level without Back Button" />
```

Reply:
>Hypothetically, we could create a non-root level without back button like this:

Thanks for the clarification, this is indeed what I had in mind. I played around with the story just to have a look and after thinking about it, I like the atomic nature of the organization. It lends itself really well to extensibility.

![Screen Shot 2020-09-09 at 11 24 00 AM](https://user-images.githubusercontent.com/1922453/92537140-8131ef80-f28f-11ea-986f-50a2f7d6af6c.png)

Some further work will be needed to get the headings correct because the main "level" can have a heading, as well as each `NavigationLevel`. Looking at the screenshot below, it appears the composibilty of the setup could lend itself to solving that problem at the consumer level.

<img width="549" alt="image" src="https://user-images.githubusercontent.com/191598/91755038-78a72e00-eb98-11ea-800c-2075e314b85c.png">